### PR TITLE
Add permissions to operator's ClusterRole that it tries to grant to some collectors

### PR DESCRIPTION
**Pull request by zarbis**
Fixes [opentelemetry-operator/issues/2824](https://github.com/open-telemetry/opentelemetry-operator/issues/2824).

Since introduction of `--create-rbac-permissions` Operator is required to have permissions it tries to grant ro Collectors. Example [code](https://github.com/open-telemetry/opentelemetry-operator/blob/c65629b3b9fcb722e770eb58b73ea4f5cc8bc63f/internal/manifests/collector/parser/processor/processor_k8sattributes.go#L78-L102).

This PR adds `extensions/replicasets` permissions unconditionally to mirror `apps/replicasets` and `nodes + namespaces` permissions only when `--create-rbac-permissions` is enabled.

### CODE REVIEW

**Comment by Allex1** [comment]
@zarbis thanks for your contribution , please follow the steps in our [guide](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/CONTRIBUTING.md) in order for the ci to pass.

**Review by Allex1 — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
This seems related to https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1123.  I think ew should wait for https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1127 to be merged.

File: charts/opentelemetry-operator/templates/clusterrole.yaml [hunk 1]
```diff
@@ -86,6 +87,15 @@ rules:
       - patch
       - update
       - watch
+  - apiGroups:
+      - ""
+    resources:
+      - nodes
+      - namespaces
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/templates/clusterrole.yaml:94]: @zarbis I see where `nodes` is being added [here](https://github.com/open-telemetry/opentelemetry-operator/blob/c65629b3b9fcb722e770eb58b73ea4f5cc8bc63f/internal/manifests/collector/parser/processor/processor_k8sattributes.go#L78-L102), but I don't see `namespaces`.

Comment (zarbis) [reply charts/opentelemetry-operator/templates/clusterrole.yaml:94]: @TylerHelmuth a bit [above](https://github.com/open-telemetry/opentelemetry-operator/blob/c65629b3b9fcb722e770eb58b73ea4f5cc8bc63f/internal/manifests/collector/parser/processor/processor_k8sattributes.go#L56-L60).

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@zarbis please resolve conflicts

**Comment by zarbis** [comment]
@TylerHelmuth done

**Comment by zarbis** [comment]
@TylerHelmuth sorry, tripped over... now done :)

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**